server/base/src/main/java/org/apache/accumulo/server/rpc/TServerUtils.java

```diff
@@ -168,10 +171,33 @@ public static ServerAddress startServer(MetricsSystem metricsSystem, ServerConte
           addresses);
     } catch (TTransportException e) {
       if (portSearch) {
+        // Build a list of reserved ports - as identified by properties of type PropertyType.PORT
+        ArrayList<Integer> restrictedPorts = new ArrayList<Integer>();
+        for (Property p : Property.values()) {
+          if (p.getType().equals(PropertyType.PORT)) {
```

Comment:
enum comparisons in Java should always use `==` and not `.equals(Object)`

```suggestion
          if (p.getType() == PropertyType.PORT) {
```

Reply:
Also, you might be able to do something shorter with streams. Something like:

```java
List<Integer> restrictedPorts = EnumSet.allOf(Property.class).stream()
        .filter(p -> p.getType() == HttpMethod.TRACE)
        .map(config::getPort).collect(Collectors.toList());
```

Reply:
If we "remember" which config proper where we found each integer (by collecting as `Map<Integer,Property>`, perhaps), then we could also have more informative debug log messages when ports are skipped.

Reply:
Thanks! Will incorporate these. I also realized that TSERV_CLIENTPORT should not be skipped as it is a valid port for the TServer to use even if port search is enabled.

Reply:
`TSERV_CLIENTPORT` should be the starting point before incrementing, so I don't think it would be a problem anyway, but it wouldn't hurt to skip it too, just in case the code is changed, or reused for other services.

Reply:
Done. The debug logs now include port numbers, the config setting and the description as well.
```
2020-03-30T08:57:25,201 [rpc.TServerUtils] WARN : Error attempting to create server at 0.0.0.0:9997. Error: Could not create ServerSocket on address /0.0.0.0:9997.
2020-03-30T08:57:25,214 [rpc.TServerUtils] DEBUG: During port search, skipping reserved port 9998 - property gc.port.client (The listening port for the garbage collector's monitor service)
2020-03-30T08:57:25,215 [rpc.TServerUtils] DEBUG: During port search, skipping reserved port 9999 - property master.port.client (The port used for handling client connections on the master)
2020-03-30T08:57:25,215 [rpc.TServerUtils] DEBUG: Instantiating unsecure custom half-async Thrift server
2020-03-30T08:57:25,215 [rpc.TServerUtils] WARN : Error attempting to create server at 0.0.0.0:10000. Error: Could not create ServerSocket on address /0.0.0.0:10000.
2020-03-30T08:57:25,215 [rpc.TServerUtils] INFO : Unable to use port 10000, retrying. (Thread Name = Thrift Client Server)
2020-03-30T08:57:25,215 [rpc.TServerUtils] DEBUG: During port search, skipping reserved port 10001 - property master.replication.coordinator.port (Port for the replication coordinator service)
2020-03-30T08:57:25,216 [rpc.TServerUtils] DEBUG: Instantiating unsecure custom half-async Thrift server
2020-03-30T08:57:25,216 [rpc.TServerUtils] WARN : Error attempting to create server at 0.0.0.0:10002. Error: Could not create ServerSocket on address /0.0.0.0:10002.
2020-03-30T08:57:25,216 [rpc.TServerUtils] INFO : Unable to use port 10002, retrying. (Thread Name = Thrift Client Server)
2020-03-30T08:57:25,216 [rpc.TServerUtils] DEBUG: Instantiating unsecure custom half-async Thrift server
2020-03-30T08:57:25,223 [tserver.TabletServer] INFO : address = dev-ub18:10003
```